packages/flutter_tools/lib/src/commands/create_base.dart

```diff
@@ -319,13 +319,40 @@ abstract class CreateBase extends FlutterCommand {
     }
   }
 
-  /// Gets the project name based.
+  /// Gets the project name.
   ///
-  /// Use the current directory path name if the `--project-name` is not specified explicitly.
+  /// If the `--project-name` is not specified explicitly,
+  /// the `name` field from the pubspec.yaml file is used.
+  ///
+  /// If the pubspec.yaml file does not exist,
+  /// the current directory path name is used.
   @protected
   String get projectName {
-    final String projectName =
-        stringArg('project-name') ?? globals.fs.path.basename(projectDirPath);
+    String? projectName = stringArg('project-name');
+
+    if (projectName == null) {
+      final File pubspec = globals.fs
+        .directory(projectDirPath)
+        .childFile('pubspec.yaml');
+
+      if (pubspec.existsSync()) {
+        final RegExp nameRegex = RegExp('^name: ([a-z0-9_]+)');
```

Comment:
@andrewkolos should we not be parsing the pubspec here, rather than just doing a regex?

Reply:
I was thinking of that, too, but isn't that a bit too much work for what is needed here?

Reply:
For example, consider the following, valid yaml:

```yaml
name:                 my_app_name
```

This would fail this regex.

Reply:
Fair point, I can either adjust the regex to include whitespace, or we can parse the YamlMap

Reply:
Added parsing the YamlMap in https://github.com/flutter/flutter/pull/150614/commits/176693e94b3a599854516d2342580d0baf694787